TACC/Core-Portal
# Feat/FP-1027 - Remove unrelated objects from app definition

## Overview: ##
This is a feature branch for FP-630. This branch reorganizes the data returned from the AppsView as suggested in this [comment](https://github.com/TACC/Core-Portal/pull/359#discussion_r592553781).

## Related Jira tickets: ##

* [FP-1027](https://jira.tacc.utexas.edu/browse/FP-1027)

## Summary of Changes: ##
The data usually returned in the app definition from `AppsView` (server/portal/apps/workspace/api/views.py) will be broken up to include the following fields:
- definition
- systemHasKeys
- pushKeysSystem
- resource
- scheduler
- exec_sys
- license
- appListing


## Testing Steps: ##
1. Try launching some jobs in the apps space or compress/decompress actions.
2. You should see a notification banner preventing you from running jobs in the apps space if you need to push keys.
